sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogManager.scala

```diff
@@ -128,6 +128,10 @@ class CatalogManager(
     }
   }
 
+  def listCatalogs(): mutable.HashMap[String, CatalogPlugin] = synchronized {
+    catalogs
```

Comment:
Only the registered (meaning the catalogs already used in the session) catalogs will be returned. Is this the desired behavior (It could be confusing to the user)?

Reply:
Yeah, This behavior is desired for me because of that it is easy to get and the displayed catalogs are meaningful because they can be loaded correctly. But at now I think it is unreasonable. From the user's point of view, when he uses this command, it should display all the catalogs that configured with conf even if its configuration is incorrect and it cannot be loaded correctly. Thank you very much @imback82 . I will rewrite this method by scanning the configuration of conf

Reply:
Catalogs are lazily loaded. Before we load them, we don't know which catalogs are available (people can set configs at runtime). This is a drawback of using configs to load catalogs, we probably should use a more explicit way to load/register a catalog.